File: app/src/test/java/com/coinninja/coinkeeper/service/tasks/LockUserTaskTest.java
package com.coinninja.coinkeeper.service.tasks;

import com.coinninja.coinkeeper.model.helpers.UserHelper;
import com.coinninja.coinkeeper.util.DateUtil;
import com.coinninja.coinkeeper.util.Intents;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;
import static org.powermock.api.mockito.PowerMockito.when;

@RunWith(MockitoJUnitRunner.class)
public class LockUserTaskTest {


    public static final long CURRENT_TIME_IN_MILLIS = 1533238943110L;


    @Mock
    private DateUtil dateUtil;

    @Mock
    private UserHelper user;

    private LockUserTask lockUserTask;


    @Before
    public void setUp() {
        when(dateUtil.getCurrentTimeInMillis()).thenReturn(CURRENT_TIME_IN_MILLIS);
        lockUserTask = new LockUserTask(user, dateUtil);
    }

    @Test
    public void locks_user_out_of_wallet_for_five_minutes() {
        lockUserTask.doInBackground();

        verify(user).lockOutUntil(CURRENT_TIME_IN_MILLIS + Intents.LOCK_DURRATION);

    }
}

File: app/src/test/java/com/coinninja/coinkeeper/service/tasks/LockUserTaskTest.kt
package com.coinninja.coinkeeper.service.tasks

import com.coinninja.coinkeeper.model.helpers.UserHelper
import com.coinninja.coinkeeper.util.DateUtil
import com.coinninja.coinkeeper.util.DropbitIntents

import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mock
import org.mockito.junit.MockitoJUnitRunner

import org.mockito.Mockito.verify
import org.mockito.Mockito.`when`

@RunWith(MockitoJUnitRunner::class)
class LockUserTaskTest {

    @Mock
    private val dateUtil: DateUtil? = null

    @Mock
    private val user: UserHelper? = null

    private var lockUserTask: LockUserTask? = null


    @Before
    fun setUp() {
        `when`(dateUtil!!.getCurrentTimeInMillis()).thenReturn(CURRENT_TIME_IN_MILLIS)
        lockUserTask = LockUserTask(user, dateUtil)
    }

    @Test
    fun locks_user_out_of_wallet_for_five_minutes() {
        lockUserTask!!.doInBackground()

        verify<UserHelper>(user).lockOutUntil(CURRENT_TIME_IN_MILLIS + DropbitIntents.LOCK_DURRATION)

    }

    companion object {

        private val CURRENT_TIME_IN_MILLIS = 1533238943110L
    }
}